index.js

```diff
@@ -129,22 +131,41 @@ module.exports = {
 
         if (opts.text) content.text = opts.text
 
-        addGroupTangle(content, (err, content) => {
+        //if (!addMemberSpec.isValid(content))
+        //  return cb(new Error(addMemberSpec.isValid.errorsString))
+
+        publish(content, (err, msg) => {
           if (err) return cb(err)
 
-          //if (!addMemberSpec.isValid(content))
-          //  return cb(new Error(addMemberSpec.isValid.errorsString))
+          //TODO: this is an optimization, use the db query instead for now
+          //keystore.group.registerAuthors(groupId, feedIds, (err) => {
+          //  if (err) return cb(err)
+          return cb(null, msg)
+          //})
+        })
+      })
+    }
+
+    function publish(content, cb) {
+      if (cb === undefined) return promisify(publish)(content)
 
-          ssb.db.create({ content, recps, encryptionFormat: 'box2' }, (err) => {
+      if (!content) return cb(new Error('Missing content'))
+
+      const recps = content.recps
+      if (!recps || !Array.isArray(recps) || recps.length < 1)
+        return cb(new Error('Missing recps'))
+
+      addGroupTangle(content, (err, content) => {
+        if (err) return cb(err)
+
+        ssb.db.create(
+          { content, recps, encryptionFormat: 'box2' },
```

Comment:
```suggestion
          { content, encryptionFormat: 'box2' },
```

Let's have only content.recps. ssb-db2 supports both ways, but I think `opts.recps` is an anti-pattern (that I accidentally created and I'll try to revert it), so let's start here by avoiding it.